Sources/Apodini/Properties/Parameter.swift

```diff
@@ -21,7 +21,13 @@ public struct Parameter<Element: Codable>: Property {
 
     
     var id = UUID()
-    var name: String?
+    var name: String? {
+        didSet {
+            if let name = name {
+                precondition(!name.isEmpty, "The name for Parameter cannot be empty!")
```

Comment:
not that it would matter (or even necessarily be an improvement over the current code), but you could do
```swift
precondition(!(str ?? " ").isEmpty, "Parameter name cannot be empty. Use `nil` to un-define a parameter name.")
```
to get rid of the if let

Reply:
Not sure if I like how the `if let` is replaced.
Also not sure if we should suggest to use `nil`, as the real solution is just to not specify a name?